File: ext/src/Column.c
/*
  Copyright (c) 2017 DataStax, Inc.

  This software can be used solely with DataStax Enterprise. Please consult the
  license at http://www.datastax.com/terms/datastax-dse-driver-license-terms
*/

#include "php_driver.h"

zend_class_entry *php_driver_column_ce = NULL;

ZEND_BEGIN_ARG_INFO_EX(arginfo_none, 0, ZEND_RETURN_VALUE, 0)
ZEND_END_ARG_INFO()

static zend_function_entry php_driver_column_methods[] = {
  PHP_ABSTRACT_ME(Column, name, arginfo_none)
  PHP_ABSTRACT_ME(Column, type, arginfo_none)
  PHP_ABSTRACT_ME(Column, isReversed, arginfo_none)
  PHP_ABSTRACT_ME(Column, isStatic, arginfo_none)
  PHP_ABSTRACT_ME(Column, isFrozen, arginfo_none)
  PHP_ABSTRACT_ME(Column, indexName, arginfo_none)
  PHP_ABSTRACT_ME(Column, indexOptions, arginfo_none)
  PHP_FE_END
};

void php_driver_define_Column(TSRMLS_D)
{
  zend_class_entry ce;

  INIT_CLASS_ENTRY(ce, PHP_DRIVER_NAMESPACE "\\Column", php_driver_column_methods);
  php_driver_column_ce = zend_register_internal_class(&ce TSRMLS_CC);
  php_driver_column_ce->ce_flags |= ZEND_ACC_INTERFACE;
}

File: ext/doc/Dse/Column.php
<?php

/*
  Copyright (c) 2016 DataStax, Inc.
  This software can be used solely with DataStax Enterprise. Please consult the
  license at http://www.datastax.com/terms/datastax-dse-driver-license-terms
*/

namespace Dse;

/**
 * A PHP representation of a column
 */
interface Column {

    /**
     * Returns the name of the column.
     *
     * @return string Name of the column or null
     */
    public function name();

    /**
     * Returns the type of the column.
     *
     * @return Type Type of the column
     */
    public function type();

    /**
     * Returns whether the column is in descending or ascending order.
     *
     * @return bool Whether the column is stored in descending order.
     */
    public function isReversed();

    /**
     * Returns true for static columns.
     *
     * @return bool Whether the column is static
     */
    public function isStatic();

    /**
     * Returns true for frozen columns.
     *
     * @return bool Whether the column is frozen
     */
    public function isFrozen();

    /**
     * Returns name of the index if defined.
     *
     * @return string Name of the index if defined or null
     */
    public function indexName();

    /**
     * Returns index options if present.
     *
     * @return string Index options if present or null
     */
    public function indexOptions();

}
